src/Language/PureScript/Errors.hs

```diff
@@ -670,7 +670,7 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage <$> onTypesInError
                                              , indent . line $ "import " ++ show im ++ " hiding (" ++ nm ++ ")"
                                              ]
     suggestions' (TypesDoNotUnify t1 t2)
-      | any isObject [t1, t2] = [line "Note that function composition in PureScript is defined using (<<<)"]
+      | isObject t1 && isFunction t2 = [line "Note that function composition in PureScript is defined using (<<<)"]
```

Comment:
Honestly, I'm not really sure. Can you give me a code example of what that might look like?